### DISCUSSION TITLE
Half-Life 1: Ray Traced 1.0.5a

### DISCUSSION THREAD

**Discussion by sultim-t** [sultim-t/xash-rt#119] [General]
## Install

1. Buy and install [Half-Life (1998)](https://store.steampowered.com/app/70/HalfLife/) on Steam
    * A clean installation without any mods is preferable
    * It doesn't work with Half-Life: Source

1. Open its folder
    * In Steam library: right click on Half Life → Manage → Browse local Files

1. Unzip `hl1-rt-combined.zip` into the folder
    * If you have a previous version of the mod, delete existing `rt` and `rt_bin` folders
    * Replace if prompted: so original configs will not interfere with HL1 RT

1. Run `rt_bin/xash3d.exe` from its folder
    * *Press **X** to switch renderers on the fly*
    * To use DLSS, additional steps are required (see below)

<details>
<summary> Optional DLSS</summary>

<br>

1. Download [RayTracedGL1-Bundle-DLSS.zip](https://github.com/sultim-t/RayTracedGL1/releases/tag/hl1-dev)
    * *Replace* the original `rt_bin/RayTracedGL1.dll` with a new one `bin/RayTracedGL1.dll` from `RayTracedGL1-Bundle-DLSS.zip`
1. **And** download [nvngx_dlss.dll](https://github.com/NVIDIA/DLSS/blob/main/lib/Windows_x86_64/rel/nvngx_dlss.dll)
    * *Place* it into the `rt_bin` folder, ensure that it's called `nvngx_dlss.dll`

</details>

<details>
<summary> Update log </summary>

<br>

* Small fix for menu artifacts

</details>

<details>
<summary>Known issues</summary>

<br>

* AMD GPUs might have some incompatibility issues
* Classic renderer doesn't support dynamic lightmaps (consequently, a flashlight)
* Classic renderer doesn't support sky visibility polygons, so some unintended objects might be observed in a skybox
* Classic renderer's decals don't have lightmap lighting
* Custom maps should work, but the lighting might be overly dark and unplayable; performance on them may be worse
* Hazard Course is not supported yet

</details>

If you would like to discuss, go to [Discussions](https://github.com/sultim-t/xash-rt/discussions)
If you have an issue, please, try find it in [Issus](https://github.com/sultim-t/xash-rt/issues) or [Discussions](https://github.com/sultim-t/xash-rt/discussions), as it might be already solved

## Download

<hr /><em>This discussion was created from the release <a href='https://github.com/sultim-t/xash-rt/releases/tag/1.0.5a'>Half-Life 1: Ray Traced 1.0.5a</a>.</em>

**Comment by Arkkis** [comment]
Barney dies at the train door. Also scientist dies in distance while train is moving. Massacre.

**Comment by red0bear** [comment]
linux version please ?

**Reply by red0bear** [reply]
proton no ... we already have native stuff .. why proton ... someone make gabe call

**Reply by palaashatri** [reply]
if it works fine, why bother?

**Comment by ghosty3354** [comment]
I know this sounds dumb but is it possible you could release a version without the rtx and just the CRT filter and new pause menu ui? I really love the CRT filter and would love to do a whole playthrough with it.

**Reply by palaashatri** [reply]
Try: https://github.com/FWGS/xash3d-fwgs

**Comment by MachaKing14** [comment]
So I've tested with RX 6600 and looks like the lighting effect is now working wonders from 1.0.5 to 1.0.5a.
Keep up the good work!

**Comment by KingSlayerBess** [comment]
Good Job

**Comment by Tonydrumz23** [comment]
Does this work with the Half life VR mod?

**Comment by nhac-lly** [comment]
If only this mod works with VR as well, it will be a thrilling experience.

**Comment by battlemonkey228** [comment]
W_loadWadFile: couldn't load gfx.wad

**Reply by groszu** [reply]
you must take the files from the hl1-rt-combined and paste them in to the main folder

**Comment by moYdaDyr** [comment]
Will there be a version of this mod on Sven Coop?

**Reply by stephen-cusi** [reply]
`no`

**Reply by moYdaDyr** [reply]
> `no`

What a shame, ok

**Comment by GreenGroove46** [comment]
I don't know why, but i have red sun shadows, and green dark, (My videocard its NVIDIA Geforce RTX 2060)
![20230421191456_1](https://user-images.githubusercontent.com/126620497/233779656-ca813b16-abf4-4585-96d1-1b8c1b2fc38f.jpg)

**Reply by GreenGroove46** [reply]
Can it be a nvidia studio driver?

**Reply by Chaek08** [reply]
> Can it be a nvidia studio driver?

Personally, I have the same problem on both game ready and studio, I think it does not depend on the type of driver

**Reply by sultim-t** [reply]
Potential fix: https://github.com/sultim-t/xash-rt/issues/191#issuecomment-1602481771

**Comment by Githubdog1** [comment]
I've got the same green lighting problem as GreenGroove46. With or without DLSS, doesn't matter. Do you guys know how to fix this?

![green lighting](https://github.com/sultim-t/xash-rt/assets/135039224/317167f3-3acd-40ba-bbf0-4312b9650938)

Thanks in advance.

**Reply by GreenGroove46** [reply]
This is maybe studio driver, try to install game ready driver.

**Reply by GreenGroove46** [reply]
I fixed it with this method.

**Reply by Githubdog1** [reply]
> This is maybe studio driver, try to install game ready driver.

I tried installing the latest NVIDIA game ready driver, but with no luck. Still the green lighting.

So any other ideas would be greatly appreciated.

**Reply by sultim-t** [reply]
Oof, maybe colors are encoded on CPU side with one encoding, but the GPU code uses another, interpreting all the lights as green... would be nice if someone would investigate, requires a deep dive though :/

**Reply by Githubdog1** [reply]
So you aren't the developer then? You know who is?

**Reply by sultim-t** [reply]
Potential fix: https://github.com/sultim-t/xash-rt/issues/191#issuecomment-1602481771

**Comment by CwavGuy** [comment]
impressive Mod! And at the very beginning I just can't believe it is the game from 1997! Thanks for your works!!!

**Comment by Githubdog1** [comment]
> Oof, maybe colors are encoded on CPU side with one encoding, but the GPU code uses another, interpreting all the lights as green... would be nice if someone would investigate, requires a deep dive though :/

So you aren't the developer of this mod? If so, do you perhaps know who is? So he can help us.

**Comment by BuClown** [comment]
Автор, а подскажи пожалуйста, возможно ли что-то похожее реализовать в аддонах(Opposing Force,Blue Shift)? Или это вообще другие игры, другой движок, другое естество реализации, или это ещё более колоссальная работа, просто так не прикрутишь?

А вообще большое человеческое спасибо за эту прекрасную реализацию, работает прекрасно, выглядит фантастически, играется одно удовольствие, вот что значит человек с руками откуда нужно, желаю всех благ и великих успехов! <3
И ещё крохотный вопросик, это реализовано просто Софтварной ТЛ? или при помощи видии и её "Ядер"? Просто у мя зелёнка 3080, и не имею информации, как это можно было бы проверить...

**Reply by sultim-t** [reply]
Спасибо, приятно!

Opposing Force, Blue Shift возможно, просто главный вопрос в ассетах, материалы, сцены. Даже в самой Half-Life 1 порядка 80-90 карт, каждую нужно настраивать вручную. Да и нужно много чего исправлять, так как в последние недели разработки хаки уже начинались, специфичные для HL1. Вопрос с лицензией на Xash также не последний.

Работает с помощью hardware ray tracing через Vulkan API, так что не через compute (software ray tracing).

**Comment by mustyyyyy** [comment]
Hey guys, is there any way to makes team count the hours when i launch the game through Xash3d???

**Comment by MattMohaupt** [comment]
Hey I am having a really strange bug where I can't move by holding down a button and it is strafing very slowly to the left and I am getting a locked 30fps on a rtx3080. When i go back to the original version(not pressing x to change render) it works perfectly fine, but cannot play this version. Any tips would be appreciated.

**Comment by gervy225** [comment]
Could this work with Hl1 MMOD?

**Reply by stephen-cusi** [reply]
no

**Comment by WYXJS** [comment]
I don’t know why, I lose my main menu

**Comment by gmpan5177** [comment]
I want to use other language sound.
How can I change it?
(sorry for my poor english)

**Comment by i-linch** [comment]
@sultim-t 
Спасибо большое за модернизацию этой замечательной игры!

После 25 лет захотелось переиграть еще раз с Вашим модом, но к сожалению не получается его установить. Помогите пожалуйста.

Итак, я вроде все сделал по инструкции: купил HL1 в Стиме, установил, скачал hl1-rt-patched.zip, разпаковал в C:\Program Files (x86)\Steam\steamapps\common\Half-Life\ и переписал при этом valve\config.cfg. Но когда я захожу в rt_bin и запускаю xash3d.exe или xash3d-console.bat, появляется абсолютно черное окно, и опять пропадает через 2 секунды. И все, больше ничего не происходит. В Half-Life я пока никаких изменений не заметил. 

В чем моя ошибка?

**Reply by XJ58** [reply]
пробовал pre 25 years anniversary качать

**Comment by Crazycatwantsrtx** [comment]
Why can't I enable NVIDIA DLSS? I have an NVIDIA GPU and stuff everything is compatible why cant i use DLSS? amd fsr is too laggy for me unlike nvidia dlss but i cant enable it

**Reply by XJ58** [reply]
is your gpu a rtx gpu

**Reply by Crazycatwantsrtx** [reply]
> is your gpu a rtx gpu

I wouldnt be complaining otherwise lolol, 2060rtx

**Comment by JSTQ** [comment]
Hello, is it possible to add fsr 3/4 to it?

**Comment by jasminaijakov-coder** [comment]
So it dosent work for linux cus i have bazzite and when i run it it says unable to find the blue noise file with / symbol but linux dosent let me rename it with / becouse " file name cant have / in it "

**Comment by lawyers3** [comment]
will you do a version that doesnt need a hardware accelerated cards, i mean like in opengl or in a not rtx version of vulkan.

**Comment by 76oy** [comment]
Я могу как то сделать, что бы этот exe запускался через steam автоматически, когда я запускаю игру?
